### ISSUE TITLE
Nightly Cuda+OpenMP builds fail to link

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos#6628]
**Describe the bug**

Nightly Cuda+OpenMP builds fail to link after recent updates (output below is for cuda/12.0+gcc/9.3.0):
```
20:19:31 CMakeFiles/Kokkos_CoreUnitTest_DeviceAndThreads.dir/UnitTest_DeviceAndThreads.cpp.o: In function `print_flag(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
20:19:31 tmpxft_0003d11a_00000000-6_UnitTest_DeviceAndThreads.cudafe1.cpp:(.text+0x7f1): undefined reference to `omp_get_max_threads'
20:19:31 CMakeFiles/Kokkos_CoreUnitTest_DeviceAndThreads.dir/UnitTest_DeviceAndThreads.cpp.o: In function `get_max_threads()':
20:19:31 tmpxft_0003d11a_00000000-6_UnitTest_DeviceAndThreads.cudafe1.cpp:(.text+0x171): undefined reference to `omp_get_max_threads'
20:19:31 ../src/libkokkoscore.a(Kokkos_OpenMP.cpp.o): In function `Kokkos::OpenMP::concurrency() const':
20:19:31 tmpxft_0003cfc0_00000000-6_Kokkos_OpenMP.cudafe1.cpp:(.text+0x180): undefined reference to `omp_get_level'
20:19:31 ../src/libkokkoscore.a(Kokkos_OpenMP.cpp.o): In function `Kokkos::OpenMP::in_parallel(Kokkos::OpenMP const&)':
20:19:31 tmpxft_0003cfc0_00000000-6_Kokkos_OpenMP.cudafe1.cpp:(.text+0x2e8): undefined reference to `omp_get_level'
20:19:31 ../src/libkokkoscore.a(Kokkos_OpenMP.cpp.o): In function `Kokkos::OpenMP::impl_thread_pool_size() const':
20:19:31 tmpxft_0003cfc0_00000000-6_Kokkos_OpenMP.cudafe1.cpp:(.text+0x310): undefined reference to `omp_get_level'
20:19:31 ../src/libkokkoscore.a(Kokkos_OpenMP.cpp.o): In function `Kokkos::OpenMP::OpenMP(int)':
20:19:31 tmpxft_0003cfc0_00000000-6_Kokkos_OpenMP.cudafe1.cpp:(.text+0x3ed): undefined reference to `omp_get_level'
20:19:31 ../src/libkokkoscore.a(Kokkos_OpenMP.cpp.o): In function `Kokkos::OpenMP::concurrency() const':
20:19:31 tmpxft_0003cfc0_00000000-6_Kokkos_OpenMP.cudafe1.cpp:(.text+0x19f): undefined reference to `omp_get_num_threads'
20:19:31 ../src/libkokkoscore.a(Kokkos_OpenMP.cpp.o): In function `Kokkos::OpenMP::impl_thread_pool_size() const':
20:19:31 tmpxft_0003cfc0_00000000-6_Kokkos_OpenMP.cudafe1.cpp:(.text+0x32f): undefined reference to `omp_get_num_threads'
20:19:31 ../src/libkokkoscore.a(Kokkos_OpenMP_Instance.cpp.o): In function `Kokkos::Impl::OpenMPInternal::clear_thread_data() [clone ._omp_fn.0]':
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0x8): undefined reference to `omp_get_thread_num'
20:19:31 ../src/libkokkoscore.a(Kokkos_OpenMP_Instance.cpp.o): In function `Kokkos::Impl::OpenMPInternal::clear_thread_data()':
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0x18d): undefined reference to `GOMP_parallel'
20:19:31 ../src/libkokkoscore.a(Kokkos_OpenMP_Instance.cpp.o): In function `Kokkos::Impl::OpenMPInternal::singleton()':
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0x43f): undefined reference to `GOMP_parallel'
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0x45b): undefined reference to `omp_get_level'
20:19:31 ../src/libkokkoscore.a(Kokkos_OpenMP_Instance.cpp.o): In function `Kokkos::Impl::OpenMPInternal::get_current_max_threads()':
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0x4d5): undefined reference to `GOMP_parallel'
20:19:31 ../src/libkokkoscore.a(Kokkos_OpenMP_Instance.cpp.o): In function `Kokkos::Impl::OpenMPInternal::initialize(int)':
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0x50d): undefined reference to `omp_in_parallel'
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0x547): undefined reference to `GOMP_parallel'
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0x592): undefined reference to `GOMP_parallel'
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0x751): undefined reference to `omp_set_num_threads'
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0x781): undefined reference to `omp_set_num_threads'
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0x7b4): undefined reference to `GOMP_parallel'
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0x7d2): undefined reference to `omp_get_level'
20:19:31 ../src/libkokkoscore.a(Kokkos_OpenMP_Instance.cpp.o): In function `Kokkos::Impl::OpenMPInternal::finalize()':
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0xc5e): undefined reference to `omp_in_parallel'
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0xd20): undefined reference to `omp_in_parallel'
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0xdb1): undefined reference to `GOMP_parallel'
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0xdcf): undefined reference to `omp_get_level'
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0xe41): undefined reference to `GOMP_parallel'
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0xe5f): undefined reference to `omp_get_level'
20:19:31 tmpxft_0003cfe9_00000000-6_Kokkos_OpenMP_Instance.cudafe1.cpp:(.text+0xec7): undefined reference to `GOMP_parallel'
20:19:31 ../src/libkokkoscore.a(Kokkos_HostSpace_deepcopy.cpp.o): In function `std::enable_if<!std::is_same<Kokkos::RangePolicy::schedule_type::type, Kokkos::Dynamic>::value, void>::type Kokkos::Impl::ParallelFor<Kokkos::Impl::hostspace_parallel_deepcopy_async(Kokkos::OpenMP const&, void*, void const*, long)::{lambda(long)#1}, Kokkos::RangePolicy<Kokkos::OpenMP>, Kokkos::OpenMP>::execute_parallel<Kokkos::RangePolicy>() const [clone ._omp_fn.0]':
20:19:31 tmpxft_0003cdee_00000000-6_Kokkos_HostSpace_deepcopy.cudafe1.cpp:(.text+0x21): undefined reference to `omp_get_num_threads'
20:19:31 tmpxft_0003cdee_00000000-6_Kokkos_HostSpace_deepcopy.cudafe1.cpp:(.text+0x28): undefined reference to `omp_get_thread_num'
20:19:31 ../src/libkokkoscore.a(Kokkos_HostSpace_deepcopy.cpp.o): In function `std::enable_if<!std::is_same<Kokkos::RangePolicy::schedule_type::type, Kokkos::Dynamic>::value, void>::type Kokkos::Impl::ParallelFor<Kokkos::Impl::hostspace_parallel_deepcopy_async(Kokkos::OpenMP const&, void*, void const*, long)::{lambda(long)#2}, Kokkos::RangePolicy<Kokkos::OpenMP>, Kokkos::OpenMP>::execute_parallel<Kokkos::RangePolicy>() const [clone ._omp_fn.0]':
20:19:31 tmpxft_0003cdee_00000000-6_Kokkos_HostSpace_deepcopy.cudafe1.cpp:(.text+0x161): undefined reference to `omp_get_num_threads'
20:19:31 tmpxft_0003cdee_00000000-6_Kokkos_HostSpace_deepcopy.cudafe1.cpp:(.text+0x168): undefined reference to `omp_get_thread_num'
20:19:31 ../src/libkokkoscore.a(Kokkos_HostSpace_deepcopy.cpp.o): In function `std::enable_if<!std::is_same<Kokkos::RangePolicy::schedule_type::type, Kokkos::Dynamic>::value, void>::type Kokkos::Impl::ParallelFor<Kokkos::Impl::hostspace_parallel_deepcopy_async(Kokkos::OpenMP const&, void*, void const*, long)::{lambda(long)#3}, Kokkos::RangePolicy<Kokkos::OpenMP>, Kokkos::OpenMP>::execute_parallel<Kokkos::RangePolicy>() const [clone ._omp_fn.0]':
20:19:31 tmpxft_0003cdee_00000000-6_Kokkos_HostSpace_deepcopy.cudafe1.cpp:(.text+0x2bb): undefined reference to `omp_get_num_threads'
20:19:31 tmpxft_0003cdee_00000000-6_Kokkos_HostSpace_deepcopy.cudafe1.cpp:(.text+0x2c3): undefined reference to `omp_get_thread_num'
20:19:31 ../src/libkokkoscore.a(Kokkos_HostSpace_deepcopy.cpp.o): In function `Kokkos::Impl::hostspace_parallel_deepcopy_async(Kokkos::OpenMP const&, void*, void const*, long)':
20:19:31 tmpxft_0003cdee_00000000-6_Kokkos_HostSpace_deepcopy.cudafe1.cpp:(.text+0x889): undefined reference to `GOMP_parallel'
20:19:31 tmpxft_0003cdee_00000000-6_Kokkos_HostSpace_deepcopy.cudafe1.cpp:(.text+0xdcc): undefined reference to `GOMP_parallel'
20:19:31 tmpxft_0003cdee_00000000-6_Kokkos_HostSpace_deepcopy.cudafe1.cpp:(.text+0x131f): undefined reference to `GOMP_parallel'
20:19:31 tmpxft_0003cdee_00000000-6_Kokkos_HostSpace_deepcopy.cudafe1.cpp:(.text+0x1571): undefined reference to `omp_get_nested'
20:19:31 tmpxft_0003cdee_00000000-6_Kokkos_HostSpace_deepcopy.cudafe1.cpp:(.text+0x1c21): undefined reference to `omp_get_nested'
20:19:31 tmpxft_0003cdee_00000000-6_Kokkos_HostSpace_deepcopy.cudafe1.cpp:(.text+0x1dc1): undefined reference to `omp_get_nested'
20:19:31 tmpxft_0003cdee_00000000-6_Kokkos_HostSpace_deepcopy.cudafe1.cpp:(.text+0x1fd1): undefined reference to `omp_get_level'
20:19:31 tmpxft_0003cdee_00000000-6_Kokkos_HostSpace_deepcopy.cudafe1.cpp:(.text+0x2051): undefined reference to `omp_get_level'
20:19:31 tmpxft_0003cdee_00000000-6_Kokkos_HostSpace_deepcopy.cudafe1.cpp:(.text+0x22c8): undefined reference to `omp_get_level'
...
```

This follows merge of these commits:

try fix ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_kokkosdev2_cuda120_gcc920_cpp17/321/changes#c60716df432ee5873886ef81505147bbe0072663) / [githubweb](https://github.com/kokkos/kokkos/commit/c60716df432ee5873886ef81505147bbe0072663))
    try fix ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_kokkosdev2_cuda120_gcc920_cpp17/321/changes#f0af4672cabb5f7a13293bb10b8b68c5191ff735) / [githubweb](https://github.com/kokkos/kokkos/commit/f0af4672cabb5f7a13293bb10b8b68c5191ff735))
    avoid pyt package ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_kokkosdev2_cuda120_gcc920_cpp17/321/changes#2779b29b556948af548eb1eae6952969e0c81e17) / [githubweb](https://github.com/kokkos/kokkos/commit/2779b29b556948af548eb1eae6952969e0c81e17))
    try ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_kokkosdev2_cuda120_gcc920_cpp17/321/changes#17af2f3c40c213958fce992ca7afb08b9ab4d963) / [githubweb](https://github.com/kokkos/kokkos/commit/17af2f3c40c213958fce992ca7afb08b9ab4d963))
    fix for macos ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_kokkosdev2_cuda120_gcc920_cpp17/321/changes#68e4bedc43cdb7ce1419b3da21fb84e797a7a295) / [githubweb](https://github.com/kokkos/kokkos/commit/68e4bedc43cdb7ce1419b3da21fb84e797a7a295))
    remove comments ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_kokkosdev2_cuda120_gcc920_cpp17/321/changes#61842b7d104402ce09a2ebd59e4657237425032b) / [githubweb](https://github.com/kokkos/kokkos/commit/61842b7d104402ce09a2ebd59e4657237425032b))
    add branching ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_kokkosdev2_cuda120_gcc920_cpp17/321/changes#374064ab75aa4fd1727781b01c8c1724470788ac) / [githubweb](https://github.com/kokkos/kokkos/commit/374064ab75aa4fd1727781b01c8c1724470788ac))
    GitHub Workflows: Use Ubuntu 22.04 instead of Fedora for Intel compiler testing ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_kokkosdev2_cuda120_gcc920_cpp17/321/changes#ae75d38951e2cebb3537649bd3e7ca046d285148) / [githubweb](https://github.com/kokkos/kokkos/commit/ae75d38951e2cebb3537649bd3e7ca046d285148))
    nvcc(wrapper): adding missing `--generate-line-info` arg ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_kokkosdev2_cuda120_gcc920_cpp17/321/changes#0262f7405e06b409df1c3a6ba0c8901ae7387110) / [githubweb](https://github.com/kokkos/kokkos/commit/0262f7405e06b409df1c3a6ba0c8901ae7387110))
    graph(HIP): adding inline keyword to fix #6623 ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_kokkosdev2_cuda120_gcc920_cpp17/321/changes#f31436a0937ba7e21acce3049d1fac2550079140) / [githubweb](https://github.com/kokkos/kokkos/commit/f31436a0937ba7e21acce3049d1fac2550079140))

Corresponding to PRs:
#6604 
#6614 
#6615 
#6624 

**Please include the following for a minimal reproducer**

Reproducer: (kokkos-dev-2)
```bash
source /projects/sems/modulefiles/utils/sems-archive-modules-init.sh
module use /home/projects/x86-64/modulefiles/local
module purge
module load sems-archive-env sems-archive-cmake/3.17.1 sems-archive-gcc/9.2.0 cuda/12.0

# generate_makefile.bash 
$KOKKOS_PATH/generate_makefile.bash --with-devices=Cuda,OpenMP --arch=Volta70 --compiler=$KOKKOS_PATH/bin/nvcc_wrapper --cxxflags="-O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized " --cxxstandard="17" --kokkos-path=$KOKKOS_PATH --with-cuda-options=enable_lambda  --no-examples   --with-options=enable_large_mem_tests

# or the cmake line
cmake -DCMAKE_CXX_COMPILER=$KOKKOS_PATH/bin/nvcc_wrapper -DCMAKE_CXX_FLAGS="-O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized" -DKokkos_ENABLE_OPENMP=ON -DKokkos_ENABLE_CUDA=ON -DKokkos_ARCH_VOLTA70=ON -DKokkos_ENABLE_TESTS=ON -DKokkos_ENABLE_EXAMPLES=OFF -DKokkos_ENABLE_LARGE_MEM_TESTS=ON -DKokkos_ENABLE_CUDA_LAMBDA=ON -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_CXX_EXTENSIONS=OFF -DCMAKE_CXX_STANDARD=17 -DCMAKE_BUILD_TYPE=RELEASE -DKokkos_ENABLE_DEPRECATION_WARNINGS=OFF -DKokkos_ENABLE_DEPRECATED_CODE_4=OFF $KOKKOS_PATH
```

**Comment by ndellingwood** [comment]
Started bisection on merge commits:

Baseline: No issues with SHA f1d466622e5004af09368a08ed6315af626ff4d6 (before the merged PRs listed above)
Problem occurs with SHA 81580ca150b54cb5c5e70b32522aa88fdc9e8747 (PR #6615)
No issues with SHA b54105701b5cae697f3d7528bbf61e9de6d9f526 (PR #6624)

Something with the updated naming options in nvcc_wrapper is triggering issues with Cuda+OpenMP builds

**Comment by ndellingwood** [comment]
This is impacting kokkos-kernels and CI testing, e.g. https://github.com/kokkos/kokkos-kernels/pull/2049

**Comment by ndellingwood** [comment]
For comparison reference, here are verbose output lines for compilation and linking steps of `UnitTest_DeviceAndThreads.cpp`

### Broken SHA https://github.com/kokkos/kokkos/commit/81580ca150b54cb5c5e70b32522aa88fdc9e8747
**Compilation**:
```
cd /ascldap/users/ndellin/kokkos/testing/NightlyFailure/core/unit_test && /ascldap/users/ndellin/kokkos/testing/NightlyFailure/../../bin/nvcc_wrapper  -DKOKKOS_DEPENDENCE -I/ascldap/users/ndellin/kokkos/tpls/gtest -I/ascldap/users/ndellin/kokkos/testing/NightlyFailure/core/unit_test -I/ascldap/users/ndellin/kokkos/core/unit_test -I/ascldap/users/ndellin/kokkos/core/unit_test/category_files -I/ascldap/users/ndellin/kokkos/testing/NightlyFailure -I/ascldap/users/ndellin/kokkos/testing/NightlyFailure/core/src -I/ascldap/users/ndellin/kokkos/core/src -I/ascldap/users/ndellin/kokkos/core/src/../../tpls/desul/include -isystem /ascldap/users/projects/x86-64/cuda/12.0/include  -O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized -O3 -DNDEBUG   -extended-lambda -Wext-lambda-captures-this -arch=sm_70 -fopenmp -std=c++17 -o CMakeFiles/Kokkos_CoreUnitTest_DeviceAndThreads.dir/UnitTest_DeviceAndThreads.cpp.o -c /ascldap/users/ndellin/kokkos/core/unit_test/UnitTest_DeviceAndThreads.cpp
```

**Linkage**:
```
/ascldap/users/ndellin/kokkos/testing/NightlyFailure/../../bin/nvcc_wrapper  -O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized -O3 -DNDEBUG  -DKOKKOS_DEPENDENCE -arch=sm_70 CMakeFiles/Kokkos_CoreUnitTest_DeviceAndThreads.dir/UnitTest_DeviceAndThreads.cpp.o  -o Kokkos_CoreUnitTest_DeviceAndThreads  -Wl,-rpath,/ascldap/users/projects/x86-64/cuda/12.0/lib64/stubs:/ascldap/users/projects/x86-64/cuda/12.0/lib64 ../src/libkokkoscore.a /ascldap/users/projects/x86-64/cuda/12.0/lib64/stubs/libcuda.so /ascldap/users/projects/x86-64/cuda/12.0/lib64/libcudart.so -ldl
```


### Clean
**Compilation**:
```
cd /ascldap/users/ndellin/kokkos/testing/NightlyFailure/core/unit_test && /ascldap/users/ndellin/kokkos/testing/NightlyFailure/../../bin/nvcc_wrapper  -DKOKKOS_DEPENDENCE -I/ascldap/users/ndellin/kokkos/tpls/gtest -I/ascldap/users/ndellin/kokkos/testing/NightlyFailure/core/unit_test -I/ascldap/users/ndellin/kokkos/core/unit_test -I/ascldap/users/ndellin/kokkos/core/unit_test/category_files -I/ascldap/users/ndellin/kokkos/testing/NightlyFailure -I/ascldap/users/ndellin/kokkos/testing/NightlyFailure/core/src -I/ascldap/users/ndellin/kokkos/core/src -I/ascldap/users/ndellin/kokkos/core/src/../../tpls/desul/include  -O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized -O3 -DNDEBUG   -extended-lambda -Wext-lambda-captures-this -arch=sm_70 -fopenmp -std=c++17 -o CMakeFiles/Kokkos_CoreUnitTest_DeviceAndThreads.dir/UnitTest_DeviceAndThreads.cpp.o -c /ascldap/users/ndellin/kokkos/core/unit_test/UnitTest_DeviceAndThreads.cpp
```

**Linkage**:
```
/ascldap/users/ndellin/kokkos/testing/NightlyFailure/../../bin/nvcc_wrapper  -O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized -O3 -DNDEBUG  -DKOKKOS_DEPENDENCE -arch=sm_70 CMakeFiles/Kokkos_CoreUnitTest_DeviceAndThreads.dir/UnitTest_DeviceAndThreads.cpp.o  -o Kokkos_CoreUnitTest_DeviceAndThreads  ../src/libkokkoscore.a /ascldap/users/projects/x86-64/cuda/12.0/lib64/stubs/libcuda.so /ascldap/users/projects/x86-64/cuda/12.0/lib64/libcudart.so -ldl /net/watson.sandia.gov/storage/fast/projects/sems/install/rhel7-x86_64/sems/compiler/gcc/9.2.0/base/lib64/libgomp.so
```

**Comment by ndellingwood** [comment]
During compilation, there is an additional `-isystem /ascldap/users/projects/x86-64/cuda/12.0/include` with https://github.com/kokkos/kokkos/commit/81580ca150b54cb5c5e70b32522aa88fdc9e8747

During linkage, there is an additional `-Wl,-rpath,/ascldap/users/projects/x86-64/cuda/12.0/lib64/stubs:/ascldap/users/projects/x86-64/cuda/12.0/lib64` and missing `/net/watson.sandia.gov/storage/fast/projects/sems/install/rhel7-x86_64/sems/compiler/gcc/9.2.0/base/lib64/libgomp.so` with https://github.com/kokkos/kokkos/commit/81580ca150b54cb5c5e70b32522aa88fdc9e8747

**Comment by fnrizzi** [comment]
looking into it with @cz4rs

**Comment by fnrizzi** [comment]
@ndellingwood https://github.com/kokkos/kokkos/pull/6629 this fixes it for me when i try on dev-2 (not sure why yet)

**Comment by ndellingwood** [comment]
Thanks @fnrizzi !

**Comment by fnrizzi** [comment]
Can you confirm it works? Just to be safe :)

**Comment by ndellingwood** [comment]
Looking at the Jenkins CI jobs, there is testing of Cuda+OpenMP with the CUDA-12.2-NVHPC job https://github.com/kokkos/kokkos/blob/0d3428087bb8270de93a64d05e6a1c91acb39c29/.jenkins#L85-L87 , I suppose this issue depends on either the OpenMP version or vendor which is why it didn't show up in CI?

**Comment by ndellingwood** [comment]
> Can you confirm it works? Just to be safe :)

@fnrizzi the build is proceeding without issue for me, thanks!

**Comment by ndellingwood** [comment]
For reference, this impacted Cuda+OpenMP nightly builds for multiple combos of cuda + gcc that were tested (cuda/11.0,1,2,4,7 and cuda/12.0 + gcc/8.3,5 and gcc/9.2)

**Comment by masterleinad** [comment]
> , I suppose this issue depends on either the OpenMP version or vendor which is why it didn't show up in CI?

We don't use a verbose `Makefile` there.

**Comment by ndellingwood** [comment]
Resolved by PR #6629 , thanks @fnrizzi for the fix!
